simphony/cuds/materials.py

```diff
@@ -0,0 +1,99 @@
+from simphony.cuds.material import Material
+
+
+class Materials(object):
+    """A class that contains materials.
+
+    Class manages the materials.
+
+    """
+    def __init__(self, uid=None, data=None, description=""):
+        self._materials = {}
+
+    def add_material(self, material):
+        """Add a material
+
+        Parameters
+        ----------
+        material : Material
+            the material to be added.
+
+        Raises
+        ------
+        ValueError:
+            If a material with the same uid already exists
+
+        """
+        try:
+            self._materials[material.uid] = Material.from_material(material)
+        except KeyError:
+            raise ValueError
+
+    def remove_material(self, uid):
+        """ Remove a material
+
+        Parameters
+        ----------
+        uid: uuid.UUID
+            the id of the material to be deleted
+
+        Raises
+        ------
+        KeyError:
+            If there is no material with the given uid
+
+        """
+        del self._materials[uid]
+
+    def update_material(self, material):
+        """ Update a material
+
+        Parameters
+        ----------
+        material : Material
+            the material to be added.
+
+        Raises
+        ------
+        ValueError:
+            If a material with this uid does not exist
+
+        """
+        self._materials[material.uid] = Material.from_material(material)
+
+    def get_material(self, uid):
```

Comment:
A comment: I think get_material is a bit cumbersome to use with the uid. It would be easier for the user to make a search with the name parameter instead. Then it would be parallel with ```engine.get_dataset``` Cf.
```
engine.materials.get_material("steel")
# or
engine.materials.get_material(material_steel.uid)
```
If there's a good reason behind using uid here, then we can ignore this comment.